axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/BudgetSupplychainService.java

```diff
@@ -105,4 +109,39 @@
     }
     return budget.getBudgetLineList();
   }
+
+  @Transactional
+  public BigDecimal computeTotalAmountCommitted(Budget budget) {
+    List<BudgetLine> budgetLineList = budget.getBudgetLineList();
+
+    if (budgetLineList == null) {
+      return BigDecimal.ZERO;
+    }
+
+    BigDecimal totalAmountCommitted = budgetLineList
+            .stream()
+            .map(BudgetLine::getAmountCommitted)
+            .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+    budget.setTotalAmountCommitted(totalAmountCommitted);
+    budgetRepository.save(budget);
+
+    return totalAmountCommitted;
+  }
+
+  public void updateBudgetLinesFromPurchaseOrder(PurchaseOrder purchaseOrder) {
+    List<PurchaseOrderLine> purchaseOrderLineList = purchaseOrder.getPurchaseOrderLineList();
+
+    if(purchaseOrderLineList == null){
+      return;
+    }
+
+    purchaseOrderLineList.forEach(purchaseOrderLine ->
```

Comment:
Maybe it's clearer to use flatMap and map here, and add null check at least for the list. I think budget is not required in db, so you must check if the budget is null too (filter(Objects::nonNull))